proxy.c

```diff
@@ -0,0 +1,133 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/socket.h>
+#include <pthread.h>
+#include <limits.h>
+#include <netdb.h>
+
+#define PORT 80
+#define BUFF_SIZE 2048
+
+void FoundHeaderHost(char* buff, char* host){
+	char* header_host;
+	header_host = strstr(buff, "Host: ");
+    if (!header_host){
+        printf("Host: header not found\n");
+        return;
+    }
+	
+	char* header_host_end = strstr(header_host, "\r\n");
+	int len = header_host_end - header_host - 6;
+	strncpy(host, header_host + 6, len);
+	host[len] = '\0';
+}
+
+int ConnectToServer(char* host) {
+    int sock_server;
+	struct sockaddr_in addr;
+
+	struct hostent* host_info = gethostbyname(host);
+	sock_server = socket(AF_INET, SOCK_STREAM, 0);
+    addr.sin_family = AF_INET;
+    addr.sin_port = htons(PORT);
+    addr.sin_addr = *((struct in_addr *)host_info->h_addr);
+
+    if (connect(sock_server, (struct sockaddr*)&addr, sizeof(addr)) < 0) {
+        perror("Connect to server error");
+        return 0;
+    }
+
+    return sock_server;
+}
+
+void* ClientFunc(void* socket_client) {
+    int sock_client = *(int*)socket_client;
+    char buff[BUFF_SIZE + 1];
+    int sock_server;
+
+	//первый запрос с хостом сервера
+    int bytes = recv(sock_client, buff, BUFF_SIZE, 0);
```

Comment:
please handle situation when request of more than 8K and contain any binary data